addons/kruise-rollout/definitions/kruise-rollout.cue

```diff
@@ -29,11 +29,13 @@ template: {
 		// use context.name as service if not filled
 		service?:           string
 		gracePeriodSeconds: *5 | int
-		// +usage=Traffic routing for ingress providers, currently only nginx is supported, later we will gradually add more types, such as Isito, Alb
-		type:               "nginx"
 		// +usage=ingress name
 		ingress: name?:     string
 	}
+	#WorkloadType: {
```

Comment:
Why should users set the workload type? If needed, it should be auto assigned value.

Reply:
For example. Too many types workload will in a helm chart. The platform also cannot know which workload you want to roll out. These two fields will help for this, and they are optional for webservice or worker it can be empty.

Reply:
The kruise rollout supported workload is limited, maybe we could directly provide the options?

Reply:
I'm not sure cue can support optional struct.

```
apiVersion: apps/v1
Kind: Deployment
or
appiVersion: kruise.api.version
kind: CloneSet

```

Reply:
WorkloadType: "Deployment" | "CloneSet"

Reply:
Since most workloads in helm chart is deployment, it's fine set deployment as default.